src/node_file.cc

```diff
@@ -2957,6 +2957,38 @@ static void Mkdtemp(const FunctionCallbackInfo<Value>& args) {
   }
 }
 
+static void MkdtempSync(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
```

Comment:
hmm, actually, why do we need a different binding? We could just change the sync branch of the original implementation to directly throw instead of passing the error down to ctx. This would avoid the repetition of most of the code here. We can probably tell if it's a sync call or not by checking the length of the arguments now.

Reply:
I personally find it a lot harder to optimize since the requirements for promise implementation, sync and callback implementations are connected a lot.

Reply:
Why would it be a lot harder to optimize? We just need to introduce another `SyncCall` helper that throws instead of setting the ctx and use that in the original implementation, and it's essentially the same as what's being done here?